tutorials/intermediate/arranging_axes.py

```diff
@@ -0,0 +1,383 @@
+
+
+##############################################################################
+# The same effect can be achieved with `~.pyplot.subplot_mosaic`,
+# but the return type is a dictionary instead of an array, where the user
+# can give the keys useful meanings.  Here we provide two lists, each list
+# representing a row, and each element in the list a key representing the
+# column.
+
+fig, axd = plt.subplot_mosaic([['upper left', 'upper right'],
+                               ['lower left', 'lower right']],
+                              figsize=(5.5, 3.5), constrained_layout=True)
+for k in axd:
+    annotate_axes(axd[k], f'axd["{k}"]', fontsize=14)
+fig.suptitle('plt.subplot_mosaic()')
+
+############################################################################
+# Axes spanning rows or columns in a grid
+# ---------------------------------------
+#
+# Sometimes we want Axes to span rows or columns of the grid.
+# There are actually multiple ways to accomplish this, but the most
+# convenient is probably to use `~.pyplot.subplot_mosaic` by repeating one
+# of the keys:
+
+fig, axd = plt.subplot_mosaic([['upper left', 'right'],
+                               ['lower left', 'right']],
+                              figsize=(5.5, 3.5), constrained_layout=True)
+for k in axd:
+    annotate_axes(axd[k], f'axd["{k}"]', fontsize=14)
+fig.suptitle('plt.subplot_mosaic()')
+
+############################################################################
+# See below for the description of how to do the same thing using
+# `~matplotlib.gridspec.GridSpec` or `~matplotlib.pyplot.subplot2grid`.
+#
+# Variable widths or heights in a grid
+# ------------------------------------
+#
+# Both `~.pyplot.subplots` and `~.pyplot.subplot_mosaic` allow the rows
+# in the grid to be different heights, and the columns to be different
+# widths using the *gridspec_kw* keyword argument.
+# Spacing parameters accepted by `~matplotlib.gridspec.GridSpec`
+# can be passed to `~matplotlib.pyplot.subplots` and
+# `~matplotlib.pyplot.subplot_mosaic`:
+
+gs_kw = dict(width_ratios=[1.4, 1], height_ratios=[1, 2])
+fig, axd = plt.subplot_mosaic([['upper left', 'right'],
+                               ['lower left', 'right']],
+                              gridspec_kw=gs_kw, figsize=(5.5, 3.5),
+                              constrained_layout=True)
+for k in axd:
+    annotate_axes(axd[k], f'axd["{k}"]', fontsize=14)
+fig.suptitle('plt.subplot_mosaic()')
+
+############################################################################
+# Nested Axes layouts
+# -------------------
+#
+# Sometimes it is helpful to have two or more grids of Axes that
+# may not need to be related to one another.  The most simple way toin
+# accomplish this is to use `.Figure.subfigures`.  Note that the subfigure
+# layouts are independent, so the Axes spines in each subfigure are not
+# necessarily aligned.   See below for a more verbose way to achieve the same
```

Comment:
There are debates about single or double space, but triple space might be a bit much.
```suggestion
# necessarily aligned.  See below for a more verbose way to achieve the same
```

Reply:
OK...